packages/webview_flutter/tizen/src/webview.cc

```diff
@@ -859,6 +860,20 @@ void WebView::HandleMethodCall(
     } else {
       result->Error("InvalidArguments", "Please set javascript string");
     }
+  } else if (method_name.compare("runJavascript") == 0) {
+    if (std::holds_alternative<std::string>(arguments)) {
+      std::string js_string = std::get<std::string>(arguments);
+      webview_instance_->EvaluateJavaScript(
+          js_string, [res = result.release()](std::string value) {
+            LOG_DEBUG("value: %s\n", value.c_str());
+            if (res) {
+              res->Success();
+              delete res;
+            }
+          });
+    } else {
+      result->Error("InvalidArguments", "Please set javascript string");
+    }
```

Comment:
How about:

```cpp
else if (method_name.compare("evaluateJavascript") == 0 ||
             method_name.compare("runJavascriptReturningResult") == 0 ||
             method_name.compare("runJavascript") == 0) {
    if (std::holds_alternative<std::string>(arguments)) {
      std::string js_string = std::get<std::string>(arguments);
      webview_instance_->EvaluateJavaScript(
          js_string, [res = result.release()](std::string value) {
            LOG_DEBUG("value: %s\n", value.c_str());
            if (res) {
              bool returns_value =
                  method_name.compare("runJavascript") == 0 ? false : true;
              if (returns_value) {
                res->Success(flutter::EncodableValue(value));
              } else {
                res->Success();
              }
              delete res;
            }
          });
    } else {
      result->Error("InvalidArguments", "Please set javascript string");
    }
  }
```

Reply:
This one is much better, thank you! :+1: